### ISSUE TITLE
Etcd related errors encountered  when creating a custom k8s 1.17 cluster .

### ISSUE THREAD

**Issue by aaronRancher** [rancher/rancher#24642]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug
Bug is not seen in `k8s v1.16.3-rancher1-1`

**Steps to reproduce (least amount of steps as possible):**
In Rancher > Click Add Cluster > Custom
Set Kubernetes Version to: `v1.17.0-rancher1-1 (experimental)`
Using jenkins job (that creates AWS nodes and and runs the registration command) to provision 3 etcd nodes, 1 control node , and 2 worker nodes.

**Result:**
Cluster is stuck provisioning. Connection fails on nodes.
![Screen Shot 2019-12-17 at 1 37 56 PM](https://user-images.githubusercontent.com/57409052/71032362-723c5900-20d2-11ea-9976-04aaee49fab3.png)
```
2019/12/17 18:24:11 [ERROR] cluster [c-9tlkd] provisioning: Failed to reconcile etcd plane: Failed to create etcd client for host [3.15.17.182]: client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 172.31.0.168:2379: connect: connection refused
```

**Other details that may be helpful:**
Provisioning a `k8s v1.16.3-rancher1-1` cluster and upgrading to `k8s v1.17.0-rancher1-1 (experimental)` works.
This is a potential timing issue.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): `v2.3-head` `version 78ed261c6`
- Installation option (single install/HA): Single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Kubernetes version (use `kubectl version`): `k8s v1.17.0-rancher1-1 (experimental)`

- Docker version (use `docker version`): 18.09
- AMI: ubuntu-16.04

**Comment by superseb** [comment]
Probably identical root cause as described in https://github.com/rancher/rancher/issues/24641#issuecomment-567051485

**Comment by aaronRancher** [comment]
The workaround described in [#24641 (comment)](https://github.com/rancher/rancher/issues/24641#issuecomment-567051485) fixes this issue in `rancher\rancher:v2.3-head` `version 5e20b72cf`

**Comment by aaronRancher** [comment]
The bug fix is verified with Rancher server `rancher\rancher:master-head` `version 7578b3c76` and with `rancher\rancher:v2.3-head` `version 7ff247737`
